server/cluster.go

```diff
@@ -115,15 +120,55 @@ func (c *RaftCluster) start() error {
 	c.cachedCluster.regionStats = newRegionStatistics(c.s.scheduleOpt, c.s.classifier)
 	c.quit = make(chan struct{})
 
-	c.wg.Add(2)
+	c.wg.Add(3)
 	go c.runCoordinator()
 	go c.runBackgroundJobs(backgroundJobInterval)
-
+	go c.runSyncRegions()
 	c.running = true
 
 	return nil
 }
 
+func (c *RaftCluster) runSyncRegions() {
+	defer logutil.LogPanic()
+	defer c.wg.Done()
+	var requests []*metapb.Region
+	// grpc has limit on message size
+	maxBatchSize := 100
+	for {
+		select {
+		case <-c.quit:
+			return
+		case first := <-c.cachedCluster.getChangedRegions():
+			requests = append(requests, first.GetMeta())
+			pending := len(c.cachedCluster.getChangedRegions())
+			for i := 0; i < pending && i < maxBatchSize; i++ {
+				region := <-c.cachedCluster.getChangedRegions()
+				requests = append(requests, region.GetMeta())
+			}
+			msg := &pdpb.MetaRegions{
+				Count:   uint32(len(requests)),
+				Regions: requests}
+			data, err := proto.Marshal(msg)
+			if err != nil {
+				log.Errorf("Report regions meet error: %s", err)
+				continue
+			}
+			req := &pdpb.SyncRegionResponse{
+				Header: &pdpb.ResponseHeader{ClusterId: c.s.clusterID},
+				Data:   data,
```

Comment:
Why marshal region metas instead of including `MetaRegions` in `SyncRegionresponse` directly?